File: VB/E2032/MainWindowViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports System
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace E2032

	Public Class MainWindowViewModel
		Private r As Random
		Protected Sub New()
			r = New Random()
			Items = New ObservableCollection(Of Item)()
			For i As Integer = 0 To 49
				Items.Add(Item.Create(i, "Item " & i, r.Next(0, 50), CType(r.Next(0, 3), Category)))
			Next i
		End Sub

		Public Overridable Property Items() As ObservableCollection(Of Item)
	End Class
	Public Class Item
		Protected Sub New()
		End Sub
		Public Shared Function Create(ByVal id As Integer, ByVal name As String, ByVal value As Double, ByVal category As Category) As Item
			Return ViewModelSource.Create(Function() New Item() With {
				.ID = id,
				.Name = name,
				.Value = value,
				.Category = category
			})
		End Function
		Public Overridable Property ID() As Integer

		Public Overridable Property Name() As String

		Public Overridable Property Value() As Double

		Public Overridable Property Category() As Category
	End Class

	Public Enum Category
		Deferred
		Normal
		Urgent
	End Enum
End Namespace

File: CS/E2032/MainWindowViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;

namespace E2032 {
    public class ViewModelBase : INotifyPropertyChanged {
        public ViewModelBase() { }

        public event PropertyChangedEventHandler PropertyChanged;
        protected virtual void OnPropertyChanged(object sender, PropertyChangedEventArgs e) {
            PropertyChanged?.Invoke(sender, e);
        }

        protected void RaisePropertyChanged(string propertyName) {
            this.OnPropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
    public class MainWindowViewModel : ViewModelBase {
        Random r;
        public MainWindowViewModel() {
            r = new Random();
            Items = new ObservableCollection<Item>();
            for (int i = 0; i < 50; i++) {
                Items.Add(new Item() { ID = i, Name = "Item " + i, Value = r.Next(0, 50), Category = (Category)r.Next(0, 3) });
            }
        }

        ObservableCollection<Item> items;
        public ObservableCollection<Item> Items {
            get {
                return items;
            }
            set {
                if (items == value) {
                    return;
                }

                items = value;
                RaisePropertyChanged(nameof(Items));
            }
        }
    }
    public class Item : ViewModelBase {
        public Item() { }

        int iD;
        public int ID {
            get {
                return iD;
            }
            set {
                if (iD == value) {
                    return;
                }

                iD = value;
                RaisePropertyChanged(nameof(ID));
            }
        }
        string name;
        public string Name {
            get {
                return name;
            }
            set {
                if (name == value) {
                    return;
                }

                name = value;
                RaisePropertyChanged(nameof(Name));
            }
        }
        double _value;
        public double Value {
            get {
                return _value;
            }
            set {
                if (_value == value) {
                    return;
                }

                _value = value;
                RaisePropertyChanged(nameof(Value));
            }
        }
        Category category;
        public Category Category {
            get {
                return category;
            }
            set {
                if (category == value) {
                    return;
                }

                category = value;
                RaisePropertyChanged(nameof(Category));
            }
        }
    }
    public enum Category {
        Deferred, Normal, Urgent
    }
}
